062-support-tiered-storage.md

```diff
@@ -0,0 +1,121 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for RemoteStorageManager and RemoteLogMetadataManager, users are able to adopt tiered storage feature on Strimzi easily.
+
+It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+### RemoteStorage
+
+Add a RemoteStorage class under storage config (under https://strimzi.io/docs/operators/latest/configuring.html#type-KafkaClusterSpec-schema-reference), which would support configuration for all tiered storage related configuration.
+
+The RemoteStorage class contains configuration for various aspects, including system config, RSM specific config, RLMM specific config.
+
+```
+kafka:
+  storage:
+    remote:
+      enabled: true
+      rlm:
+        # See below detail
+      rsm:
+        # See below detail
+      rlmm:
+        # See below detail
+```
+
+### RLM
+
+The RLM config supports `default` type only.  The config allows users to specify basic configuration for RemoteLogManager in Kafka:
+
+```
+      rlm:
+        threadPoolSize:
+        reader.threads:
+        reader.maxPendingTasks:
```

Comment:
Should this be really specified here? Can't the users use the `.spec.kafka.config` section for it if needed?

Reply:
Theoretically, user can specify everything via .spec.kafka.config. That was the original proposal for. Here the config was highlighted and configured at RLM section to make sure tiered storage config are handled in 1 place, instead of spreading in different sections.

Reply:
The way this needs to be done is that the operator takes responsibility and ownership of some _selected_ fields. For example around security, what listener should be used, how to authenticate with Kafka etc. These fields should be not available to users for any customization and are driven through the Kafka custom resource API. The rest of the configuration can live in `.spec.kafka.config` which is the place where the configuration lives in general. It makes it easier that way exactly because it leaves all the things handled in one place - the whole validation of the configuration etc. You also want to be careful about having dedicated fields in the API for various options -> they make the APi big, but also make it hard to add or remove options. The `.spec.kafka.config` is opaque map where you can easily add custom configurations without any special support.

Reply:
Understood. I revise the config to minimal now.